### PR TITLE
metadata: Add parsing of WiFi status information

### PR DESCRIPTION
**Pull request by emys-alb**
Created a function called get_wifi_data() that gathers information about the wifi status at one dictionary.

Output:

    {'addr': '50:b7:c3:c9:1c:8c',
    'channel': {'band': 2462, 'center1': 2462, 'number': 11, 'width': 20},
    'ssid': 'VIVO-81A0',
    'txpower': 15.0,
    'type': 'managed'}

### CODE REVIEW

File: flent/metadata.py [hunk 1]
```diff
@@ -514,3 +516,39 @@ def get_module_versions():
...
+
+                if k  in unwanted_keys:
+                  continue
+
+                if k == 'txpower':
+                   v = float(parts[1])
+
+                if k == 'channel':
+                    # This condition will return a dict with all the values of the channel
+		            # Output will be {'addr':..., channel': {'band': 2462, 'center1': 2462, 'number': 11, 'width': 20}, 'ssid':...}
```
Comment (tohojo) [inline flent/metadata.py]: The indentation is off on the second comment line. Also, please add an example of the input (i.e., the line you're trying to parse) along with the output.

File: flent/metadata.py [hunk 2]
```diff
@@ -514,3 +516,39 @@ def get_module_versions():
...
+                    v = {}
+                    v['number'] =int(parts[1])
+                    v['band'] = int(parts[2].strip("("));
+                    v['width'] = int(parts[5])
+                    v['center1'] = int(parts[8])
+
+                if k == "multicast TXQ":
```
Comment (tohojo) [inline flent/metadata.py]: Please add a short comment explaining this (could just be, e.g., "No interesting output after this")

File: flent/metadata.py [hunk 3]
```diff
@@ -514,3 +516,39 @@ def get_module_versions():
...
+                    break
+
+            wifi_data[k] = v
```
Comment (tohojo) [inline flent/metadata.py]: This should be inside the if statement.

Alternatively, you could save one level of indentation by reversing the if statement above and turning it into a "continue". I.e., replace

```
if len(line.strip()) >= 1:
   # lots of code here
```
with
```
if not line.strip():
  continue
# code here (unindented)
```

Up to you which you prefer :)

File: flent/metadata.py [hunk 4]
```diff
@@ -138,6 +138,7 @@ def record_metadata(results, extended, hostnames):
     if extended:
         m['IP_ADDRS'] = get_ip_addrs()
         m['GATEWAYS'] = get_gateways()
+        m['WIFI_DATA'] =get_wifi_data()
```
Comment (tohojo) [inline flent/metadata.py]: Missing space after =

**Comment by emys-alb** [comment]
The second one has some changes in the code and I wasn't sure if --amend
would amend everything or just the commit message.

**Comment by emys-alb** [comment]
And thank you very much!!

**Comment by tohojo** [comment]
git --amend will also add any new changes to the existing commit, not just change the commit message. It's a way to keep the history "clean" (having one logical commit) while making changes.

**Comment by tohojo** [comment]
Huh, I guess Github isn't smart enough to figure out that that commit was a manual merge of the PR, so it shows up as "closed" instead of "merged". But don't worry, the actual changes are merged :)

Also, I fixed up the author name on your commit to use your full name; that is good practice, so you may want to update your git config to do this by default. See https://help.github.com/en/github/using-git/setting-your-username-in-git

**Pull request closed without merging**